### PR TITLE
Add sidebar to allow navigation between examples

### PR DESCRIPTION
**Pull request by matteo95g**
Screen Shot 2021-03-12 at 11 57 35 (2)

### CODE REVIEW

File: app/helpers/sidebar_helper.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+module SidebarHelper
+  def menu_active?(name)
+    controller.controller_name == name ? 'active' : ''
+  end
+
+  def sidebar_sections
```
Comment (JuannFerrari) [inline app/helpers/sidebar_helper.rb:6]: I would move this to a constant, makes more semantic sense to me

SIDEBAR_SECTIONS and then freeze the hash. WDYT?

File: app/javascript/stylesheets/sidebar.scss [hunk 2]
```diff
@@ -0,0 +1,15 @@
+.sidebar {
+  @apply flex flex-col w-52 bg-gradient-to-b from-black to-gray-800;
+
+  .item {
+    @apply flex justify-center items-center h-10 cursor-pointer text-white font-semibold text-lg capitalize transition;
+
+    &.active {
```
Comment (JuannFerrari) [inline app/javascript/stylesheets/sidebar.scss]: Let's use BEM notation here http://getbem.com/

for example:

```
.sidebar{

 &__item{
 
  &--active{
  
  
```

File: app/javascript/stylesheets/sidebar.scss [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+      @apply border-l-8 border-rs-yellow text-opacity-80 bg-gray-700;
+    }
+
+    &:hover {
```
Comment (JuannFerrari) [inline app/javascript/stylesheets/sidebar.scss]: Also if we have a one liner we should use:

`      &:hover { @apply text-opacity-80 bg-gray-700; }`

File: app/views/shared/sidebar/_sidebar.html.erb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<div class='sidebar'>
+  <div class='text-white w-3/4 self-center my-10'>
+    <svg xmlns='http://www.w3.org/2000/svg' id='Layer_1' x='0' y='0' version='1.1' viewBox='0 0 369.9 37.5' xml:space='preserve' fill='currentColor'>
```
Comment (JuannFerrari) [inline app/views/shared/sidebar/_sidebar.html.erb]: We should store this as an asset and then show it in the view using the Rails helper

**Review by JuannFerrari — COMMENTED** [review]
Great job! love it

File: app/helpers/sidebar_helper.rb (hunk 1 above)
Comment (matteo95g) [reply app/helpers/sidebar_helper.rb:6]: tried, but returns an error:

```undefined local variable or method `boards_path' for SidebarHelper:Module (NameError)```

Also tried including `url_helpers` at the beginning of the module without success:

`include Rails.application.routes.url_helpers`

I'm leaving it as it was, but I added the `.freeze` method

File: app/helpers/svg_helper.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+module SvgHelper
```
Comment (JuannFerrari) [inline app/helpers/svg_helper.rb]: I like this, but there is a gem that could be useful in the future too.

https://github.com/jamesmartin/inline_svg

Sorry I didn't comment earlier

**Review by JuannFerrari — APPROVED** [review]

File: app/views/shared/sidebar/_sidebar.html.erb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<div class='sidebar'>
+  <div class='text-white w-3/4 self-center my-10'>
+    <%= inline_svg_tag 'logo.svg' %>
+  </div>
+  <% sidebar_sections.each do |section| %>
+    <%= render 'shared/sidebar/sidebar_item', name: section[:name], path: section[:path], sections: sidebar_sections %>
```
Comment (sebaherrera07) [inline app/views/shared/sidebar/_sidebar.html.erb]: is it necessary to pass `sections: sidebar_sections` param?

**Review by sebaherrera07 — APPROVED** [review]

File: app/views/shared/sidebar/_sidebar.html.erb (hunk 6 above)
Comment (matteo95g) [reply app/views/shared/sidebar/_sidebar.html.erb]: nope, it wasn't, fixed!

File: app/helpers/sidebar_helper.rb [hunk 7]
```diff
@@ -0,0 +1,18 @@
+module SidebarHelper
+  def menu_active?(name)
```
Comment (mwolman) [inline app/helpers/sidebar_helper.rb:2]:
```suggestion
  def menu_active(name)
```

https://rubystyle.guide/#bool-methods-qmark

File: app/views/shared/sidebar/_sidebar.html.erb [hunk 8]
```diff
@@ -0,0 +1,8 @@
+<div class='sidebar'>
+  <div class='text-white w-3/4 self-center my-10'>
+    <%= inline_svg_tag 'logo.svg' %>
+  </div>
+  <% sidebar_sections.each do |section| %>
```
Comment (mwolman) [inline app/views/shared/sidebar/_sidebar.html.erb:5]: NIT
```suggestion
  
  <% sidebar_sections.each do |section| %>
```

File: tailwind.config.js [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+module.exports = {
+  purge: [],
+  darkMode: false, // or 'media' or 'class'
+  theme: {
+    extend: {},
+    colors: {
+      transparent: 'transparent',
+      current: 'currentColor',
+
```
Comment (mwolman) [inline tailwind.config.js:11]:
```suggestion
```

File: tailwind.config.js [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+      yellow: colors.amber,
+      green: colors.emerald,
+      blue: colors.blue,
+      indigo: colors.indigo,
+      purple: colors.violet,
+      pink: colors.pink,
+      'rs-gray': {
+        100: '#F1F2F5',
+        200: '#B5B5B8',
+        300: '#808183'
```
Comment (mwolman) [inline tailwind.config.js:25]:
```suggestion
        300: '#808183',
```

**Pull request merged**